src/components/PopupWithForm.js

```diff
@@ -3,21 +3,31 @@ import Popup from "./Popup.js";
 export default class PopupWithForm extends Popup {
   #submitHandler;
   #form;
+  #submitButton;
 
   constructor(popupSelector, submitHandler) {
     super(popupSelector);
     this.#submitHandler = submitHandler;
     this.#form = this.modalElement.querySelector(".modal__form");
+    this.#submitButton = this.#form.querySelector(".modal__save-button");
   }
 
   #getInputValues() {
-    this.inputInfo = {};
+    const inputInfo = {};
     const inputs = Array.from(this.modalElement.querySelectorAll(".modal__input"));
     inputs.forEach(input => {
-      this.inputInfo[input.placeholder] = input.value;
+      inputInfo[input.name] = input.value;
     })
 
-    return this.inputInfo;
+    return inputInfo;
+  }
+
+  changeButtonText (isLoading) {
+    if (isLoading) {
+      this.#submitButton.textContent = "Saving...";
+    } else {
+      this.#submitButton.textContent = "Save";
```

Comment:
**Might be better**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```